props/proptypes.md

```diff
@@ -0,0 +1,226 @@
+
+However, when another component renders a `Student` component, but doesn't pass in the `name` and `email` properties, our `Student` component looks totally broken!
+
+<!-- prettier-ignore-start -->
+```js
+const StudentList = () => {
+    return (
+        <section>
+                <Student></Student>
+                <Student name='Soo-ah' email='sooah@dev.org'></Student>
+        </section>
+    )
+}
+```
+<!-- prettier-ignore-end -->
+
+![An attendance web app with a list of two students. One student reads "Nickname: , Email: ." The second student reads "Nickname: Soo-ah, Email: sooah@dev.org"](../assets/props_proptypes_undefined.png)  
+__Fig. An attendance web app with a list of two students. One student reads "Nickname: , Email: ." The second student reads "Nickname: Soo-ah, Email: sooah@dev.org"_
+
+It would be nice if the React library could enforce the presence or data type of our props...
+
+## PropTypes
+
+**PropTypes** is a React feature that allows us to explicitly list the props a component takes and their types.
+
+When we use PropTypes effectively, PropTypes will:
+
+- Provide type-checking and validation of props
+- Create implied self-documentation of our components
+
+Both of these benefit our developer workflow and communication!
+
+## PropType Syntax
+
+We define PropTypes per component.
+
+For each component, we need to do the following:
+
+1. Import `PropTypes` into our component file
+1. Attach a `propTypes` member to our component function
+1. Assign the `propTypes` member to an object, where each key-value pair is a `prop` name and a _validator_
+
+Here, we are adding PropTypes to a component `ComponentA`.
+
+In this example, these PropTypes state that there are two expected `props` for `ComponentA`:
+
+- `title`, which is a string, and required
+- `operate`, which is a function
+
+<!-- prettier-ignore-start -->
+```js
+import PropTypes from 'prop-types';
+
+const ComponentA = (props) => {
+    return (
+        <span>{props.title}</span>
+    )
+}
+
+ComponentA.propTypes = {
+    title: PropTypes.string.isRequired,
+    operate: PropTypes.func
+};
+
+export default ComponentA;
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                                          |
+| --------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| `import PropTypes from 'prop-types';`               | We need to import `PropTypes` properly before we use it.                                                                                       |
+| `ComponentA.propTypes`                              | After and outside the `ComponentA` function definition, we will set a value to the `propTypes` member of `ComponentA`.                         |
```

Comment:
make specific mention about why PropTypes itself needs to be imported (the name)
assign work better here (😆)
equate this strange assignment to something like a static/class method
```suggestion
| `import PropTypes from 'prop-types';`               | We need to import `PropTypes` properly before we use it. Notice that we use the `PropTypes` name literally in each validator we set up, which is why we must import `PropTypes` specifically.                                                                                      |
| `ComponentA.propTypes`                              | After and outside the `ComponentA` function definition, we will assign a value to a new `propTypes` member of `ComponentA`. Note that this is a member of the entire component function (like a class member) rather than belonging to any particular component instance.                        |
```

Reply:
Simon: 0, The word "assign": 40928348235

You can tell that I even tried to get it right with this one!! hahahahaha

Reply:
I could practically hear the second-guessing going on in this sentence! 😭